src/psbt.js

```diff
@@ -231,6 +234,135 @@ export function psbtOutputFormatter(output) {
   }
 }
 
+/**
+ * Translates a PSBT into inputs/outputs consumable by non-PSBT devices in the
+ * `unchained-wallets` library.
+ *
+ * FIXME - Have only confirmed this is working for P2SH addresses on Ledger on regtest
+ *
+ * @param {module:networks.NETWORKS} network - bitcoin network
+ * @param {module:multisig.MULTISIG_ADDRESS_TYPES} addressType - address type
+ * @param {String} psbt - PSBT as a base64 or hex string
+ * @param {Object} keyDetails - Object containing signing key details (Fingerprint + bip32 prefix)
+ * @returns {null|Object} returns object with the format
+ * {
+ *    txInputs: [],
+ *    txOutputs: [],
+ *    bip32Derivations: [],
+ * }
+ */
+export function translatePSBT(network, addressType, psbt, keyDetails) {
+  let localPSBT = autoLoadPSBT(psbt, {network: networkData(network)});
+  if (localPSBT === null) return null;
+
+  let bip32Derivations = [];
+  // To provide the non-PSBT wallet with the proper info, we actually
+  // need to grab some info from the "data inputs" and some other
+  // information from the "transaction inputs".
+  let txInputs = localPSBT.txInputs.map((input, index) => {
+    const dataInput = localPSBT.data.inputs[index];
+    // The bip32Derivation hanging off of the dataInput contains
+    // an array of objects like:
+    //    [
+    //      { masterFingerprint1, pubkey1, path1 },
+    //      { masterFingerprint2, pubkey2, path2 },
```

Comment:
`masterFingerprint` is the same as `rootFingerprint` and `xfp`?

Reply:
Yes